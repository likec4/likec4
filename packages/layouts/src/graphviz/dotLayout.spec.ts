import { describe, expect, it } from 'vitest'
import {
  computedAmazonView,
  computedCloudView,
  computedCloud3levels,
  computedIndexView
} from './__fixtures__'
import { dotLayout } from './dotLayout'

describe('dotLayout:', () => {
  it('computedIndexView', async () => {
    const diagram = await dotLayout(computedIndexView)
    expect(diagram).toMatchSnapshot()
  })

  it('computedAmazonView', async () => {
    const diagram = await dotLayout(computedAmazonView)
    expect(diagram).toMatchSnapshot()
  })

  it('computedCloud3levels', async () => {
    const diagram = await dotLayout(computedCloud3levels)
    expect(diagram).toMatchSnapshot()
  })

  it('computedCloudView', async () => {
    const diagram = await dotLayout(computedCloudView)
    expect(diagram).toMatchSnapshot()
  })
})
