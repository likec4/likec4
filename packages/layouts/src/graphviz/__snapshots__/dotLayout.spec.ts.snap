// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`dotLayout: > computedAmazonView 1`] = `
{
  "autoLayout": "TB",
  "boundingBox": {
    "height": 597,
    "width": 395,
    "x": 0,
    "y": 0,
  },
  "description": null,
  "edges": [
    {
      "headArrow": [
        [
          201,
          366,
        ],
        [
          198,
          376,
        ],
        [
          195,
          366,
        ],
      ],
      "id": "cloud.backend:amazon.s3",
      "label": "persists artifacts",
      "labels": [
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            202,
            310,
          ],
          "text": "persists artifacts",
          "width": 97,
        },
      ],
      "parent": null,
      "points": [
        [
          198,
          243,
        ],
        [
          198,
          282,
        ],
        [
          198,
          327,
        ],
        [
          198,
          366,
        ],
      ],
      "relations": [
        "cloud.backend.storage:amazon.s3",
      ],
      "source": "cloud.backend",
      "target": "amazon.s3",
    },
  ],
  "height": 597,
  "id": "amazon",
  "links": null,
  "nodes": [
    {
      "children": [
        "cloud.backend",
      ],
      "color": "primary",
      "description": null,
      "id": "cloud",
      "kind": "system",
      "labels": [
        {
          "align": "left",
          "fontSize": 12,
          "fontStyle": "bold",
          "pt": [
            11,
            15,
          ],
          "text": "CLOUD",
          "width": 43,
        },
      ],
      "links": null,
      "navigateTo": "cloudView",
      "parent": null,
      "position": [
        11,
        11,
      ],
      "shape": "rectangle",
      "size": {
        "height": 259,
        "width": 373,
      },
      "tags": null,
      "technology": null,
      "title": "cloud",
    },
    {
      "children": [
        "amazon.s3",
      ],
      "color": "primary",
      "description": "Amazon is a cloud provider",
      "id": "amazon",
      "kind": "system",
      "labels": [
        {
          "align": "left",
          "fontSize": 12,
          "fontStyle": "bold",
          "pt": [
            11,
            15,
          ],
          "text": "AMAZON",
          "width": 53,
        },
      ],
      "links": null,
      "parent": null,
      "position": [
        11,
        328,
      ],
      "shape": "rectangle",
      "size": {
        "height": 259,
        "width": 373,
      },
      "tags": null,
      "technology": null,
      "title": "Amazon",
    },
    {
      "children": [],
      "color": "primary",
      "description": null,
      "id": "cloud.backend",
      "kind": "component",
      "labels": [
        {
          "align": "left",
          "fontSize": 18,
          "pt": [
            126,
            95,
          ],
          "text": "Backend",
          "width": 68,
        },
      ],
      "links": null,
      "parent": "cloud",
      "position": [
        38,
        63,
      ],
      "shape": "rectangle",
      "size": {
        "height": 180,
        "width": 320,
      },
      "tags": null,
      "technology": null,
      "title": "Backend",
    },
    {
      "children": [],
      "color": "primary",
      "description": "S3 is a storage service",
      "id": "amazon.s3",
      "kind": "component",
      "labels": [
        {
          "align": "left",
          "fontSize": 18,
          "pt": [
            149,
            83,
          ],
          "text": "S3",
          "width": 22,
        },
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            92,
            108,
          ],
          "text": "S3 is a storage service",
          "width": 135,
        },
      ],
      "links": null,
      "parent": "amazon",
      "position": [
        38,
        380,
      ],
      "shape": "rectangle",
      "size": {
        "height": 180,
        "width": 320,
      },
      "tags": null,
      "technology": null,
      "title": "S3",
    },
  ],
  "rules": [
    {
      "exprs": [
        {
          "wildcard": true,
        },
        {
          "element": "cloud",
          "isDescedants": false,
        },
        {
          "source": {
            "element": "cloud",
            "isDescedants": true,
          },
          "target": {
            "element": "amazon",
            "isDescedants": false,
          },
        },
      ],
      "isInclude": true,
    },
  ],
  "tags": null,
  "title": "",
  "viewOf": "amazon",
  "width": 395,
}
`;

exports[`dotLayout: > computedCloud3levels 1`] = `
{
  "autoLayout": "TB",
  "boundingBox": {
    "height": 1459,
    "width": 842,
    "x": 0,
    "y": 0,
  },
  "description": null,
  "edges": [
    {
      "headArrow": [
        [
          361,
          933,
        ],
        [
          358,
          942,
        ],
        [
          355,
          933,
        ],
      ],
      "id": "cloud.backend.graphql:cloud.backend.storage",
      "label": null,
      "labels": [],
      "parent": "cloud.backend",
      "points": [
        [
          358,
          844,
        ],
        [
          358,
          872,
        ],
        [
          358,
          904,
        ],
        [
          358,
          933,
        ],
      ],
      "relations": [
        "cloud.backend.graphql:cloud.backend.storage",
      ],
      "source": "cloud.backend.graphql",
      "target": "cloud.backend.storage",
    },
    {
      "headArrow": [
        [
          441,
          656,
        ],
        [
          433,
          661,
        ],
        [
          436,
          652,
        ],
      ],
      "id": "cloud.frontend.adminPanel:cloud.backend.graphql",
      "label": "fetches data in zero trust network with sso authentification",
      "labels": [
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            514,
            578,
          ],
          "text": "fetches data in zero trust network with",
          "width": 224,
        },
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            514,
            594,
          ],
          "text": "sso authentification",
          "width": 115,
        },
      ],
      "parent": "cloud",
      "points": [
        [
          558,
          511,
        ],
        [
          521,
          555,
        ],
        [
          477,
          608,
        ],
        [
          440,
          653,
        ],
      ],
      "relations": [
        "cloud.frontend.adminPanel:cloud.backend.graphql",
      ],
      "source": "cloud.frontend.adminPanel",
      "target": "cloud.backend.graphql",
    },
    {
      "headArrow": [
        [
          316,
          651,
        ],
        [
          317,
          661,
        ],
        [
          310,
          654,
        ],
      ],
      "id": "cloud.frontend.dashboard:cloud.backend.graphql",
      "label": "fetches data",
      "labels": [
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            294,
            586,
          ],
          "text": "fetches data",
          "width": 73,
        },
      ],
      "parent": "cloud",
      "points": [
        [
          251,
          511,
        ],
        [
          270,
          555,
        ],
        [
          293,
          607,
        ],
        [
          313,
          652,
        ],
      ],
      "relations": [
        "cloud.frontend.dashboard:cloud.backend.graphql",
      ],
      "source": "cloud.frontend.dashboard",
      "target": "cloud.backend.graphql",
    },
    {
      "headArrow": [
        [
          214,
          317,
        ],
        [
          211,
          326,
        ],
        [
          208,
          317,
        ],
      ],
      "id": "customer:cloud.frontend.dashboard",
      "label": "opens in browser",
      "labels": [
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            215,
            247,
          ],
          "text": "opens in browser",
          "width": 102,
        },
      ],
      "parent": null,
      "points": [
        [
          211,
          181,
        ],
        [
          211,
          223,
        ],
        [
          211,
          273,
        ],
        [
          211,
          317,
        ],
      ],
      "relations": [
        "customer:cloud.frontend.dashboard",
      ],
      "source": "customer",
      "target": "cloud.frontend.dashboard",
    },
    {
      "headArrow": [
        [
          634,
          317,
        ],
        [
          631,
          326,
        ],
        [
          628,
          317,
        ],
      ],
      "id": "support:cloud.frontend.adminPanel",
      "label": "manages",
      "labels": [
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            635,
            247,
          ],
          "text": "manages",
          "width": 55,
        },
      ],
      "parent": null,
      "points": [
        [
          631,
          181,
        ],
        [
          631,
          223,
        ],
        [
          631,
          273,
        ],
        [
          631,
          317,
        ],
      ],
      "relations": [
        "support:cloud.frontend.adminPanel",
      ],
      "source": "support",
      "target": "cloud.frontend.adminPanel",
    },
    {
      "headArrow": [
        [
          361,
          1265,
        ],
        [
          358,
          1274,
        ],
        [
          355,
          1265,
        ],
      ],
      "id": "cloud.backend.storage:amazon",
      "label": "persists artifacts",
      "labels": [
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            362,
            1220,
          ],
          "text": "persists artifacts",
          "width": 97,
        },
      ],
      "parent": null,
      "points": [
        [
          358,
          1127,
        ],
        [
          358,
          1170,
        ],
        [
          358,
          1221,
        ],
        [
          358,
          1265,
        ],
      ],
      "relations": [
        "cloud.backend.storage:amazon.s3",
      ],
      "source": "cloud.backend.storage",
      "target": "amazon",
    },
  ],
  "height": 1459,
  "id": "cloud3levels",
  "links": null,
  "nodes": [
    {
      "children": [
        "cloud.frontend.adminPanel",
        "cloud.frontend.dashboard",
        "cloud.backend",
      ],
      "color": "primary",
      "description": null,
      "id": "cloud",
      "kind": "system",
      "labels": [
        {
          "align": "left",
          "fontSize": 12,
          "fontStyle": "bold",
          "pt": [
            11,
            15,
          ],
          "text": "CLOUD",
          "width": 43,
        },
      ],
      "links": null,
      "navigateTo": "cloudView",
      "parent": null,
      "position": [
        11,
        266,
      ],
      "shape": "rectangle",
      "size": {
        "height": 927,
        "width": 820,
      },
      "tags": null,
      "technology": null,
      "title": "cloud",
    },
    {
      "children": [
        "cloud.backend.graphql",
        "cloud.backend.storage",
      ],
      "color": "primary",
      "description": null,
      "id": "cloud.backend",
      "kind": "component",
      "labels": [
        {
          "align": "left",
          "fontSize": 12,
          "fontStyle": "bold",
          "pt": [
            11,
            15,
          ],
          "text": "BACKEND",
          "width": 60,
        },
      ],
      "links": null,
      "parent": "cloud",
      "position": [
        171,
        612,
      ],
      "shape": "rectangle",
      "size": {
        "height": 541,
        "width": 373,
      },
      "tags": null,
      "technology": null,
      "title": "Backend",
    },
    {
      "children": [],
      "color": "primary",
      "description": "Component that allows to query data via GraphQL.",
      "id": "cloud.backend.graphql",
      "kind": "component",
      "labels": [
        {
          "align": "left",
          "fontSize": 18,
          "pt": [
            113,
            75,
          ],
          "text": "Graphql API",
          "width": 95,
        },
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            41,
            100,
          ],
          "text": "Component that allows to query data via",
          "width": 238,
        },
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            131,
            116,
          ],
          "text": "GraphQL.",
          "width": 59,
        },
      ],
      "links": null,
      "parent": "cloud.backend",
      "position": [
        198,
        664,
      ],
      "shape": "rectangle",
      "size": {
        "height": 180,
        "width": 320,
      },
      "tags": null,
      "technology": null,
      "title": "Graphql API",
    },
    {
      "children": [],
      "color": "primary",
      "description": "The backend storage is a component that stores data.",
      "id": "cloud.backend.storage",
      "kind": "component",
      "labels": [
        {
          "align": "left",
          "fontSize": 18,
          "pt": [
            93,
            75,
          ],
          "text": "Backend Storage",
          "width": 133,
        },
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            37,
            100,
          ],
          "text": "The backend storage is a component that",
          "width": 246,
        },
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            125,
            116,
          ],
          "text": "stores data.",
          "width": 70,
        },
      ],
      "links": null,
      "parent": "cloud.backend",
      "position": [
        198,
        947,
      ],
      "shape": "storage",
      "size": {
        "height": 180,
        "width": 320,
      },
      "tags": null,
      "technology": null,
      "title": "Backend Storage",
    },
    {
      "children": [],
      "color": "primary",
      "description": "The admin panel is a webapp that allows support staff to manage customer data.",
      "id": "cloud.frontend.adminPanel",
      "kind": "component",
      "labels": [
        {
          "align": "left",
          "fontSize": 18,
          "pt": [
            76,
            75,
          ],
          "text": "Admin Panel Webapp",
          "width": 168,
        },
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            40,
            100,
          ],
          "text": "The admin panel is a webapp that allows",
          "width": 241,
        },
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            44,
            116,
          ],
          "text": "support staff to manage customer data.",
          "width": 232,
        },
      ],
      "links": null,
      "parent": "cloud",
      "position": [
        471,
        331,
      ],
      "shape": "rectangle",
      "size": {
        "height": 180,
        "width": 320,
      },
      "tags": null,
      "technology": null,
      "title": "Admin Panel Webapp",
    },
    {
      "children": [],
      "color": "primary",
      "description": "The customer dashboard is a webapp that allows customers to view their data.",
      "id": "cloud.frontend.dashboard",
      "kind": "component",
      "labels": [
        {
          "align": "left",
          "fontSize": 18,
          "pt": [
            43,
            75,
          ],
          "text": "Customer Dashboard Webapp",
          "width": 235,
        },
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            36,
            100,
          ],
          "text": "The customer dashboard is a webapp that",
          "width": 250,
        },
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            55,
            116,
          ],
          "text": "allows customers to view their data.",
          "width": 211,
        },
      ],
      "links": null,
      "parent": "cloud",
      "position": [
        51,
        331,
      ],
      "shape": "rectangle",
      "size": {
        "height": 180,
        "width": 320,
      },
      "tags": null,
      "technology": null,
      "title": "Customer Dashboard Webapp",
    },
    {
      "children": [],
      "color": "primary",
      "description": null,
      "id": "customer",
      "kind": "actor",
      "labels": [
        {
          "align": "left",
          "fontSize": 18,
          "pt": [
            125,
            96,
          ],
          "text": "customer",
          "width": 72,
        },
      ],
      "links": null,
      "parent": null,
      "position": [
        51,
        0,
      ],
      "shape": "person",
      "size": {
        "height": 180,
        "width": 320,
      },
      "tags": null,
      "technology": null,
      "title": "customer",
    },
    {
      "children": [],
      "color": "primary",
      "description": "Support engineers are responsible for supporting customers",
      "id": "support",
      "kind": "actor",
      "labels": [
        {
          "align": "left",
          "fontSize": 18,
          "pt": [
            93,
            76,
          ],
          "text": "Support Engineer",
          "width": 135,
        },
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            49,
            101,
          ],
          "text": "Support engineers are responsible for",
          "width": 224,
        },
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            96,
            117,
          ],
          "text": "supporting customers",
          "width": 128,
        },
      ],
      "links": null,
      "parent": null,
      "position": [
        471,
        0,
      ],
      "shape": "person",
      "size": {
        "height": 180,
        "width": 320,
      },
      "tags": null,
      "technology": null,
      "title": "Support Engineer",
    },
    {
      "children": [],
      "color": "primary",
      "description": "Amazon is a cloud provider",
      "id": "amazon",
      "kind": "system",
      "labels": [
        {
          "align": "left",
          "fontSize": 18,
          "pt": [
            128,
            83,
          ],
          "text": "Amazon",
          "width": 64,
        },
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            79,
            108,
          ],
          "text": "Amazon is a cloud provider",
          "width": 161,
        },
      ],
      "links": null,
      "navigateTo": "amazon",
      "parent": null,
      "position": [
        198,
        1279,
      ],
      "shape": "rectangle",
      "size": {
        "height": 180,
        "width": 320,
      },
      "tags": null,
      "technology": null,
      "title": "Amazon",
    },
  ],
  "rules": [
    {
      "exprs": [
        {
          "wildcard": true,
        },
        {
          "element": "cloud.frontend",
          "isDescedants": true,
        },
        {
          "element": "cloud.backend",
          "isDescedants": true,
        },
      ],
      "isInclude": true,
    },
    {
      "exprs": [
        {
          "element": "cloud.frontend",
          "isDescedants": false,
        },
      ],
      "isInclude": false,
    },
  ],
  "tags": null,
  "title": "",
  "viewOf": "cloud",
  "width": 842,
}
`;

exports[`dotLayout: > computedCloudView 1`] = `
{
  "autoLayout": "TB",
  "boundingBox": {
    "height": 1391,
    "width": 680,
    "x": 0,
    "y": 0,
  },
  "description": null,
  "edges": [
    {
      "headArrow": [
        [
          344,
          892,
        ],
        [
          340,
          901,
        ],
        [
          337,
          892,
        ],
      ],
      "id": "cloud.frontend:cloud.backend",
      "label": "fetches data",
      "labels": [
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            344,
            847,
          ],
          "text": "fetches data",
          "width": 73,
        },
      ],
      "parent": "cloud",
      "points": [
        [
          340,
          781,
        ],
        [
          340,
          816,
        ],
        [
          340,
          856,
        ],
        [
          340,
          892,
        ],
      ],
      "relations": [
        "cloud.frontend.dashboard:cloud.backend.graphql",
        "cloud.frontend.adminPanel:cloud.backend.graphql",
      ],
      "source": "cloud.frontend",
      "target": "cloud.backend",
    },
    {
      "headArrow": [
        [
          286,
          588,
        ],
        [
          288,
          597,
        ],
        [
          281,
          591,
        ],
      ],
      "id": "customer:cloud.frontend",
      "label": "opens in browser",
      "labels": [
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            255,
            530,
          ],
          "text": "opens in browser",
          "width": 102,
        },
      ],
      "parent": null,
      "points": [
        [
          212,
          464,
        ],
        [
          234,
          503,
        ],
        [
          260,
          549,
        ],
        [
          283,
          589,
        ],
      ],
      "relations": [
        "customer:cloud.frontend.dashboard",
      ],
      "source": "customer",
      "target": "cloud.frontend",
    },
    {
      "headArrow": [
        [
          373,
          589,
        ],
        [
          367,
          597,
        ],
        [
          367,
          587,
        ],
      ],
      "id": "support:cloud.frontend",
      "label": "manages",
      "labels": [
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            467,
            377,
          ],
          "text": "manages",
          "width": 55,
        },
      ],
      "parent": null,
      "points": [
        [
          494,
          181,
        ],
        [
          461,
          289,
        ],
        [
          406,
          473,
        ],
        [
          371,
          587,
        ],
      ],
      "relations": [
        "support:cloud.frontend.adminPanel",
      ],
      "source": "support",
      "target": "cloud.frontend",
    },
    {
      "headArrow": [
        [
          344,
          1197,
        ],
        [
          340,
          1207,
        ],
        [
          337,
          1197,
        ],
      ],
      "id": "cloud.backend:amazon",
      "label": "persists artifacts",
      "labels": [
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            344,
            1152,
          ],
          "text": "persists artifacts",
          "width": 97,
        },
      ],
      "parent": null,
      "points": [
        [
          340,
          1086,
        ],
        [
          340,
          1121,
        ],
        [
          340,
          1161,
        ],
        [
          340,
          1197,
        ],
      ],
      "relations": [
        "cloud.backend.storage:amazon.s3",
      ],
      "source": "cloud.backend",
      "target": "amazon",
    },
  ],
  "height": 1391,
  "id": "cloudView",
  "links": null,
  "nodes": [
    {
      "children": [
        "cloud.frontend",
        "cloud.backend",
      ],
      "color": "primary",
      "description": null,
      "id": "cloud",
      "kind": "system",
      "labels": [
        {
          "align": "left",
          "fontSize": 12,
          "fontStyle": "bold",
          "pt": [
            10,
            15,
          ],
          "text": "CLOUD",
          "width": 43,
        },
      ],
      "links": null,
      "navigateTo": "cloud3levels",
      "parent": null,
      "position": [
        154,
        548,
      ],
      "shape": "rectangle",
      "size": {
        "height": 564,
        "width": 373,
      },
      "tags": null,
      "technology": null,
      "title": "cloud",
    },
    {
      "children": [],
      "color": "primary",
      "description": null,
      "id": "cloud.frontend",
      "kind": "component",
      "labels": [
        {
          "align": "left",
          "fontSize": 18,
          "pt": [
            126,
            95,
          ],
          "text": "Frontend",
          "width": 70,
        },
      ],
      "links": null,
      "parent": "cloud",
      "position": [
        180,
        600,
      ],
      "shape": "browser",
      "size": {
        "height": 180,
        "width": 320,
      },
      "tags": null,
      "technology": null,
      "title": "Frontend",
    },
    {
      "children": [],
      "color": "primary",
      "description": null,
      "id": "cloud.backend",
      "kind": "component",
      "labels": [
        {
          "align": "left",
          "fontSize": 18,
          "pt": [
            127,
            95,
          ],
          "text": "Backend",
          "width": 68,
        },
      ],
      "links": null,
      "parent": "cloud",
      "position": [
        180,
        906,
      ],
      "shape": "rectangle",
      "size": {
        "height": 180,
        "width": 320,
      },
      "tags": null,
      "technology": null,
      "title": "Backend",
    },
    {
      "children": [],
      "color": "primary",
      "description": null,
      "id": "customer",
      "kind": "actor",
      "labels": [
        {
          "align": "left",
          "fontSize": 18,
          "pt": [
            125,
            95,
          ],
          "text": "customer",
          "width": 72,
        },
      ],
      "links": null,
      "parent": null,
      "position": [
        0,
        283,
      ],
      "shape": "person",
      "size": {
        "height": 180,
        "width": 320,
      },
      "tags": null,
      "technology": null,
      "title": "customer",
    },
    {
      "children": [],
      "color": "primary",
      "description": "Support engineers are responsible for supporting customers",
      "id": "support",
      "kind": "actor",
      "labels": [
        {
          "align": "left",
          "fontSize": 18,
          "pt": [
            93,
            76,
          ],
          "text": "Support Engineer",
          "width": 135,
        },
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            49,
            101,
          ],
          "text": "Support engineers are responsible for",
          "width": 224,
        },
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            97,
            117,
          ],
          "text": "supporting customers",
          "width": 128,
        },
      ],
      "links": null,
      "parent": null,
      "position": [
        360,
        0,
      ],
      "shape": "person",
      "size": {
        "height": 180,
        "width": 320,
      },
      "tags": null,
      "technology": null,
      "title": "Support Engineer",
    },
    {
      "children": [],
      "color": "primary",
      "description": "Amazon is a cloud provider",
      "id": "amazon",
      "kind": "system",
      "labels": [
        {
          "align": "left",
          "fontSize": 18,
          "pt": [
            129,
            83,
          ],
          "text": "Amazon",
          "width": 64,
        },
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            80,
            108,
          ],
          "text": "Amazon is a cloud provider",
          "width": 161,
        },
      ],
      "links": null,
      "navigateTo": "amazon",
      "parent": null,
      "position": [
        180,
        1211,
      ],
      "shape": "rectangle",
      "size": {
        "height": 180,
        "width": 320,
      },
      "tags": null,
      "technology": null,
      "title": "Amazon",
    },
  ],
  "rules": [
    {
      "exprs": [
        {
          "wildcard": true,
        },
      ],
      "isInclude": true,
    },
  ],
  "tags": null,
  "title": "",
  "viewOf": "cloud",
  "width": 680,
}
`;

exports[`dotLayout: > computedIndexView 1`] = `
{
  "autoLayout": "TB",
  "boundingBox": {
    "height": 1074,
    "width": 680,
    "x": 0,
    "y": 0,
  },
  "description": null,
  "edges": [
    {
      "headArrow": [
        [
          284,
          576,
        ],
        [
          286,
          585,
        ],
        [
          279,
          579,
        ],
      ],
      "id": "customer:cloud",
      "label": "opens in browser",
      "labels": [
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            260,
            530,
          ],
          "text": "opens in browser",
          "width": 102,
        },
      ],
      "parent": null,
      "points": [
        [
          214,
          464,
        ],
        [
          235,
          499,
        ],
        [
          259,
          540,
        ],
        [
          281,
          577,
        ],
      ],
      "relations": [
        "customer:cloud.frontend.dashboard",
      ],
      "source": "customer",
      "target": "cloud",
    },
    {
      "headArrow": [
        [
          379,
          577,
        ],
        [
          373,
          585,
        ],
        [
          373,
          575,
        ],
      ],
      "id": "support:cloud",
      "label": "manages",
      "labels": [
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            470,
            377,
          ],
          "text": "manages",
          "width": 55,
        },
      ],
      "parent": null,
      "points": [
        [
          496,
          181,
        ],
        [
          471,
          271,
        ],
        [
          430,
          415,
        ],
        [
          390,
          538,
        ],
        [
          386,
          550,
        ],
        [
          382,
          563,
        ],
        [
          377,
          575,
        ],
      ],
      "relations": [
        "support:cloud.frontend.adminPanel",
      ],
      "source": "support",
      "target": "cloud",
    },
    {
      "headArrow": [
        [
          344,
          880,
        ],
        [
          340,
          889,
        ],
        [
          337,
          880,
        ],
      ],
      "id": "cloud:amazon",
      "label": "persists artifacts",
      "labels": [
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            344,
            835,
          ],
          "text": "persists artifacts",
          "width": 97,
        },
      ],
      "parent": null,
      "points": [
        [
          340,
          769,
        ],
        [
          340,
          804,
        ],
        [
          340,
          844,
        ],
        [
          340,
          880,
        ],
      ],
      "relations": [
        "cloud.backend.storage:amazon.s3",
      ],
      "source": "cloud",
      "target": "amazon",
    },
  ],
  "height": 1074,
  "id": "index",
  "links": null,
  "nodes": [
    {
      "children": [],
      "color": "primary",
      "description": null,
      "id": "customer",
      "kind": "actor",
      "labels": [
        {
          "align": "left",
          "fontSize": 18,
          "pt": [
            125,
            95,
          ],
          "text": "customer",
          "width": 72,
        },
      ],
      "links": null,
      "parent": null,
      "position": [
        0,
        283,
      ],
      "shape": "person",
      "size": {
        "height": 180,
        "width": 320,
      },
      "tags": null,
      "technology": null,
      "title": "customer",
    },
    {
      "children": [],
      "color": "primary",
      "description": null,
      "id": "cloud",
      "kind": "system",
      "labels": [
        {
          "align": "left",
          "fontSize": 18,
          "pt": [
            140,
            96,
          ],
          "text": "cloud",
          "width": 42,
        },
      ],
      "links": null,
      "navigateTo": "cloudView",
      "parent": null,
      "position": [
        180,
        588,
      ],
      "shape": "rectangle",
      "size": {
        "height": 180,
        "width": 320,
      },
      "tags": null,
      "technology": null,
      "title": "cloud",
    },
    {
      "children": [],
      "color": "primary",
      "description": "Support engineers are responsible for supporting customers",
      "id": "support",
      "kind": "actor",
      "labels": [
        {
          "align": "left",
          "fontSize": 18,
          "pt": [
            93,
            76,
          ],
          "text": "Support Engineer",
          "width": 135,
        },
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            49,
            101,
          ],
          "text": "Support engineers are responsible for",
          "width": 224,
        },
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            97,
            117,
          ],
          "text": "supporting customers",
          "width": 128,
        },
      ],
      "links": null,
      "parent": null,
      "position": [
        360,
        0,
      ],
      "shape": "person",
      "size": {
        "height": 180,
        "width": 320,
      },
      "tags": null,
      "technology": null,
      "title": "Support Engineer",
    },
    {
      "children": [],
      "color": "primary",
      "description": "Amazon is a cloud provider",
      "id": "amazon",
      "kind": "system",
      "labels": [
        {
          "align": "left",
          "fontSize": 18,
          "pt": [
            129,
            83,
          ],
          "text": "Amazon",
          "width": 64,
        },
        {
          "align": "left",
          "fontSize": 14,
          "pt": [
            80,
            108,
          ],
          "text": "Amazon is a cloud provider",
          "width": 161,
        },
      ],
      "links": null,
      "navigateTo": "amazon",
      "parent": null,
      "position": [
        180,
        894,
      ],
      "shape": "rectangle",
      "size": {
        "height": 180,
        "width": 320,
      },
      "tags": null,
      "technology": null,
      "title": "Amazon",
    },
  ],
  "rules": [
    {
      "exprs": [
        {
          "wildcard": true,
        },
      ],
      "isInclude": true,
    },
  ],
  "tags": null,
  "title": "",
  "width": 680,
}
`;
